src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.impl.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private OperationStrategy strategy;
+    private HashMap storage;
+
+    public ShopServiceImpl() {
+        strategy = new OperationStrategyImpl();
+        storage = new HashMap<String, Integer>();
```

Comment:
The `storage` field should be defined with an interface type such as `Map` instead of a concrete implementation `HashMap` to follow the principle of coding to an interface rather than an implementation.